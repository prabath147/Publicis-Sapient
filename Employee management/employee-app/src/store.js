import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./Component/reducers";
import Thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(Thunk))
);

export default store;
