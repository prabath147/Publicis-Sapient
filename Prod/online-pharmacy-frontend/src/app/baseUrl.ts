export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
