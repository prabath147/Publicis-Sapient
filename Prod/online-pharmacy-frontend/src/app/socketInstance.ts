
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);