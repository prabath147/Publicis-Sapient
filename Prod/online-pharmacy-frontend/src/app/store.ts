import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
