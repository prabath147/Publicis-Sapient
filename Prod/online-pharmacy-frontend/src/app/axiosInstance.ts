import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
