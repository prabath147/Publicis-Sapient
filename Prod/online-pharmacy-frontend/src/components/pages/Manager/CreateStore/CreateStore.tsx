import { Box, Button, Group, Modal, TextInput, Title, useMantineTheme } from '@mantine/core';
import { useForm } from '@mantine/form';
import { useAppDispatch, useAppSelector } from '../../../../app/hooks';
import { getUserData } from '../../Auth/Login/UserSlice';
import { createNewStore } from '../ManagerStoreList/ManagerStoreListAPI';

interface ModalStateType {
  open: boolean,
  setOpen: (boolean) => void
}

export default function CreateStore({ open, setOpen }: ModalStateType) {
  // const user = LocalUserService.getLocalUser();
  const user = useAppSelector(getUserData);
  const dispatch = useAppDispatch();
  const theme = useMantineTheme();
  const form = useForm({
    initialValues: {
      storeName: '',
      street: '',
      city: '',
      state: '',
      country: '',
      pinCode: 0
    },

    validate: {
      storeName: (value) => (value.length === 0 ? 'store name cannot be empty' : null),
      street: (value) => (value.length === 0 ? 'street name cannot be empty' : null),
      city: (value) => (value.length === 0 ? 'city name cannot be empty' : null),
      state: (value) => (value.length === 0 ? 'state name cannot be empty' : null),
      country: (value) => (value.length === 0 ? 'country name cannot be empty' : null),
      pinCode: (value) => (value.toString().length <= 5 ? 'invalid pin code' : null),
    },
  });

  const handleSubmit = (values: any) => {
    console.log(values);
    const data = {
      address: {
        city: values.city,
        country: values.country,
        pinCode: values.pinCode,
        state: values.state,
        street: values.street
      },
      createdDate: (new Date()).toISOString().slice(0, 10),
      manager: {
        managerId: user.id
      },
      revenue: 0,
      storeName: values.storeName
    };
    dispatch(createNewStore(data));
    closeModal();
  }
  const closeModal = () => {
    setOpen(false);
    form.reset();
  }
  return (
    <Modal
      opened={open}
      onClose={closeModal}
      title="Create Store"
      overlayColor={theme.colorScheme === 'dark' ? theme.colors.dark[9] : theme.colors.gray[2]}
      overlayOpacity={0.55}
      overlayBlur={3}
    >

      <Box >
        <form onSubmit={form.onSubmit(handleSubmit)}>
          <TextInput
            withAsterisk
            label="Store Name"
            placeholder="name"
            {...form.getInputProps('storeName')}
          />
          <Title my={15} order={6}>Store Address</Title>
          <TextInput
            withAsterisk
            label="Street"
            placeholder="street"
            {...form.getInputProps('street')}
            mt={4}
          />
          <TextInput
            withAsterisk
            label="City"
            placeholder="city"
            {...form.getInputProps('city')}
            mt={4}
          />
          <TextInput
            withAsterisk
            label="State"
            placeholder="state"
            {...form.getInputProps('state')}
            mt={4}
          />
          <TextInput
            withAsterisk
            label="Country"
            placeholder="country"
            {...form.getInputProps('country')}
            mt={4}
          />
          <TextInput
            withAsterisk
            label="PIN Code"
            placeholder="******"
            {...form.getInputProps('pinCode')}
            mt={4}
          />
          <Group position="right" mt="md">
            <Button fullWidth type="submit">Create</Button>
          </Group>
        </form>
      </Box>
    </Modal>
  );
}
