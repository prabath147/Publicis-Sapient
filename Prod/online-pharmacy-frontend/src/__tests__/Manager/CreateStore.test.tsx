import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-test-renderer';
import { store } from '../../app/store';
import CreateStore from '../../components/pages/Manager/CreateStore/CreateStore';
import mockAxios from '../../__mocks__/axios';


const MockCreateStore = () => {
    return <Provider store={store}>
        <BrowserRouter>
            <CreateStore open={true} setOpen={() => true} />
        </BrowserRouter>
    </Provider>
}

describe('component created', () => {
    afterEach(() => {
        jest.resetAllMocks();
    });

    mockAxios.post.mockImplementationOnce(() =>
        Promise.resolve()
    )

    it('should render the component', () => {
        const { asFragment } = render(<MockCreateStore />)
        expect(asFragment()).toMatchSnapshot();
    })

    it('should render the handle create call', async () => {
        const { asFragment, getByLabelText, getByRole } = render(<MockCreateStore />)

        // await act(async () => {
        //     fireEvent.click(getByRole('button', { name: /Add Store/i }));
        // })
        expect(asFragment()).toMatchSnapshot()

        await act(async () => {
            fireEvent.change(getByLabelText(/Store Name/i), { target: { value: "name1" } })

            fireEvent.change(getByLabelText(/Street/i), { target: { value: "s1" } })
            fireEvent.change(getByLabelText(/City/i), { target: { value: "c2" } })
            fireEvent.change(getByLabelText(/State/i), { target: { value: "s1" } })
            fireEvent.change(getByLabelText(/Country/i), { target: { value: "c1" } })
            fireEvent.change(getByLabelText(/PIN Code/i), { target: { value: "123456" } })
        })

        await act(async () => {
            fireEvent.click(getByRole('button', { name: /Create/i }));
        })
        expect(asFragment()).toMatchSnapshot()
    })
})
