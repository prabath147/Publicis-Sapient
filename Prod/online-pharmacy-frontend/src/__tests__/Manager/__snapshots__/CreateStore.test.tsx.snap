// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component created should render the component 1`] = `<DocumentFragment />`;

exports[`component created should render the handle create call 1`] = `<DocumentFragment />`;

exports[`component created should render the handle create call 2`] = `<DocumentFragment />`;
